### PR TITLE
Sort mgr sync products

### PR DESCRIPTION
**Pull request by mcalmer**
## What does this PR change?

Sort the product list in mgr-sync output

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- No documentation needed: only internal and user invisible changes

- [x] **DONE**

## Test coverage
- No tests: already covered

- [x] **DONE**

## Links

Tracks # **add downstream PR, if any**

- [ ] **DONE**

## Changelogs

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_oracle"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

### CODE REVIEW

File: susemanager/src/mgr_sync/product.py [hunk 1]
```diff
@@ -39,6 +40,21 @@ def __init__(self, data):
...
+
+    def __ne__(self, other):
+        return not (self == other)
+
+    def __lt__(self, other):
+        selfIsSUSE = self.friendly_name.startswith("SUSE")
+        otherIsSUSE = other.friendly_name.startswith("SUSE")
+        if selfIsSUSE and not otherIsSUSE:
+            return False
+        if not selfIsSUSE and otherIsSUSE:
```
Comment (agraul) [inline susemanager/src/mgr_sync/product.py:54]: [nitpick]: I would prefer snake_case to be consistent with Python's style:
```suggestion
        self_is_SUSE = self.friendly_name.startswith("SUSE")
        other_is_SUSE = other.friendly_name.startswith("SUSE")
        if self_is_SUSE and not other_is_SUSE:
            return False
        if not self_is_SUSE and other_is_SUSE:
```

**Review by agraul — APPROVED** [review]
Looks good in general, I only have a small nitpick. 

As always, feel free to ignore my nitpicks :).

File: susemanager/src/mgr_sync/product.py [hunk 2]
```diff
@@ -39,6 +40,21 @@ def __init__(self, data):
     def __repr__(self):
         return self.to_ascii_row()
 
+    def __eq__(self, other):
+        return (self.friendly_name == other.friendly_name)
+
+    def __ne__(self, other):
+        return not (self == other)
```
Comment (agraul) [inline susemanager/src/mgr_sync/product.py:47]: As far as I know this is redundant, the default for `__ne__` is already to take the inverse of `__eq__` ([doc](https://docs.python.org/3/reference/datamodel.html#object.__ne__))

Comment (mcalmer) [reply susemanager/src/mgr_sync/product.py:47]: I copied an example and they also said this, but implemented it anyway :-) I will try to remove it

Comment (mcalmer) [reply susemanager/src/mgr_sync/product.py]: removed

**Pull request merged**